papers/Vvmax/calc_vvmax.py

```diff
@@ -0,0 +1,764 @@
+        
+        #if False:
+        plt.figure()
+        plt.plot(ztrials,effz,label="efficiency")
+        plt.plot(ztrials,effz/eff0,label="Rel efficiency")
+        plt.plot(ztrials,modFz,label="Relative lum dist")
+        plt.plot(ztrials,sz/s0,label="product")
+        plt.plot([zmax,zmax],[0.,1./s0],color="black",linestyle=":")
+        plt.plot([ztrials[0],zmax],[1./s0,1./s0],color="black",linestyle=":")
+        
+        plt.xlabel("z guesses")
+        plt.ylabel("Relative efficiency factors")
+        plt.yscale('log')
+        #plt.ylim(0.1,10)
+        plt.xlim(0.0001,0.1)
+        plt.xscale('log')
+        plt.legend()
+        plt.tight_layout()
+        plt.savefig("example_zguesses.pdf")
+        plt.close()
+        exit()
+    
+    return zmax
+    
+def zefficiency(z0,w0,DM0,freq,tsamp,newz,newDM):
+    """
+    Calculates width if the burst had been at a higher redshift
+    Returns efficiency: propto w**-0.5
+    This modifies detectable fluence
+    
+    The new DM and z must be input by the user
+    """
+    
+    # calculates width components at z0
+    dmsmear0 = DM0 * 4.15 * 1e6 * ((freq-0.5)**-2 - (freq+0.5)**-2)
+    
+    wintrinsic = (w0**2 - dmsmear0**2 - tsamp**2)
+    if wintrinsic < 0.:
+        wintrinsic = 0.
+    else:
+        wintrinsic = wintrinsic**0.5
+    
+    new_dmsmear = newDM * 4.15 * 1e6 * ((freq-0.5)**-2 - (freq+0.5)**-2)
+    
+    # reduces intrinsic width as 1+z
+    new_int = wintrinsic * (1.+z0)/(1.+newz)
+    new_w = ((new_int)**2 + new_dmsmear**2 + tsamp**2)**0.5
+    eff = new_w **-0.5
+    return eff
+
+
+def dvdtau(z):
+    """ Comoving volume element [Mpc^3 dz sr^-1].
+    normalsied by proper time (1+z)^-1 to convert
+    rates.
+    """ 
+    iz=np.array(np.floor(z/DZ)).astype('int')
+    kz=z/DZ-iz
+    return (dvdtaus[iz]*(1.-kz)+dvdtaus[iz+1]*kz) #removed the 1/(1+z) dependency
+
+
+def F_to_E(F,z,alpha,bandwidth=1e9):
+    """ Converts a fluence to an energy
+    Formula from Macquart & Ekers 2018
+    Fluence assumed to be in Jy ms
+    Energy is returned in ergs (here, erg/Hz)
+    """
+    E = F * (4*np.pi*(cos.dl(z))**2/(1.+z)**(2.+alpha))
+    E *= 9.523396e22*bandwidth 
+    return E
+
+def E_to_F(E,z,alpha, bandwidth=1e9):
+    """ Converts an energy to a fluence
+    Formula from Macquart & Ekers 2018
+    Energy is assumed to be in ergs
+    Fluence returned in Jy ms
+    """
+    F=E/(4*np.pi*(cos.dl(z))**2/(1.+z)**(2.+alpha))
+    F /= 9.523396e22*bandwidth # see below for constant calculation
+    return F
+  
+
+def plot_fluence(DM,Fluences,Fnu):
+    """
+    Creates a plot of the "Fluences" calculated by Clancy 
+    vs the "Fnu" calculated by Wayne
+    """
+    plt.figure()
+    plt.scatter(DM, Fluences/Fnu)
+    plt.xlabel("DM")
+    plt.ylabel("James F / Arcus F")
+    plt.ylim(0,2)
+    #plt.xscale("log")
+    #plt.yscale("log")
+    plt.savefig("frb_fluences.pdf")
+    plt.close()
+
+def calc_fluence(F0,Bvals,SNR,w,tsamp,DM,freq,Nants):
```

Comment:
holy cow, this module is large

maybe break it up if you move code from it into `zdm`

Reply:
I think I've appropriately broken up useful functions from this in the main module, under vvmax.py.